lab-jaren/lib/parse-message.js

```diff
@@ -0,0 +1,49 @@
+'use strict';
+
+module.exports = exports = {};
+
+exports.nickCommand = function(message, socket) {
+  socket.username = message.split(' ').slice(1).join(' ').trim();
+  socket.write(`you are now ${socket.username}\n`);
+};
+
+exports.dmCommand = function(message, array) {
+  let toUsername = message.split(' ')[1];
+  let content = message.split(' ').slice(2).join(' ').trim();
+  array.forEach(user => {
+    if (user.username === toUsername)
+      user.write(content + '\n');
+  });
+};
+
+exports.usersCommand = function(array, socket) {
+  array.forEach(user => {
+    socket.write(user.username + '\n');
+  });
+};
+
+exports.trollCommand = function(message, array) {
+  let content = message.split(' ').slice(1).join(' ').trim();
+  array.forEach(user => {
+    for (var i = 0; i < 10; i++){
+      user.write(content + '\n');
+    }
+  });
+};
+
+exports.banCommand = function(message, array) {
```

Comment:
Remember your ``/ban`` command needs to delete a user from your sockets array too.